app/src/processing/app/Base.java

```diff
@@ -1585,6 +1640,21 @@ public void actionPerformed(ActionEvent actionevent) {
 
     JRadioButtonMenuItem item = new JRadioButtonMenuItem(action);
 
+    // create an action for the "recent boards" copy of this menu item
+    // which clicks the original menu item
+    Action actionClone = new AbstractAction(board.getName()) {
+      public void actionPerformed(ActionEvent actionevent) {
+        item.setSelected(true);
+        item.getAction().actionPerformed(new ActionEvent(this, -1, ""));
+      }
+    }; 
+
+    // create a menu item for the "recent boards" menu
+    JMenuItem itemClone = new JMenuItem(actionClone);
+
+    // populate list of menuitem copies
+    recentBoardItems.put(boardId, itemClone);
```

Comment:
The problem I have is that during `createBoardMenusAndCustomMenus()` I don't know what is going to be on the recent boards list, and the list is updated more often than `createBoardMenusAndCustomMenus()` is called. Whether I save the item, a copy of the item, or just the action, I need to save one for every board that's installed. There are only two solutions I can imagine:

1) Add a function that drills down through all installed packages and calls `createBoardMenusAndCustomMenus()`. This new function would probably be called in `rebuildRecentBoardsList()`. I think this would significantly slow down switching boards or ports so probably not a good idea.

2) Instead of cloning and saving the menu items, retrieve them from the menu during `rebuildRecentBoardsList()`. Presumably, they live somewhere? I spent a long time trying to figure out how to do this... never did.

Reply:
> Whether I save the item, a copy of the item, or just the action, I need to save one for every board that's installed.

Yes, I was suggesting you still have a map that contains an item for each board in the entire menu, but you wouldn't have to create the clone item and action for each, just for the ones you're actually going to use.

You could probably also find them without the map, but just using the map is probably a lot faster then trying to find them from the menus.

Reply:
I implemented this locally, cleans up things nicely. However, some other small improvements made broke things, so I haven't pushed this anywhere yet, no time to fix that now, so I'll come back to this. Also the custom board menu stuff might also need some further changes here.

Reply:
I implemented this in my branch https://github.com/matthijskooijman/Arduino/commit/a9631da193a4d0325020d753c310a29d2ccfc421

Note that when stuff is refactored by saving `fqbn`'s, as I suggest in #10887, this might end up different again, though.